### PR TITLE
Migrate pipelines and profiling code to `ST::string`

### PR DESCRIPTION
**Pull request by dgelessus**
Just some cleanup to replace more C-style string manipulation and fixed-size buffers. This should have no visible effect.

Almost all of these are "boring" changes. The only somewhat complex parts are the number formatting in `plProfileBase` and the graphics adapter/driver name checks in `hsG3DDeviceSelector`.

### CODE REVIEW

**Comment by Hoikas** [comment]
Due to the changes to pfDXPipeline, this will be deferred until after #1326 is merged.

File: Sources/Plasma/PubUtilLib/plStatGather/plProfileManagerFull.cpp [hunk 1]
```diff
@@ -73,11 +73,8 @@ void plProfileManagerFull::GetGroups(GroupSet& groups)
         groups.insert(fVars[i]->GetGroup());
 }
 
-void plProfileManagerFull::ShowGroup(const char* groupName)
+void plProfileManagerFull::ShowGroup(const ST::string& groupName)
 {
-    if (!groupName)
-        groupName = "General";
```
[the remark this answers is not in the document]
Comment (dgelessus) [reply Sources/Plasma/PubUtilLib/plStatGather/plProfileManagerFull.cpp:79]: Nothing, like for any other group that doesn't exist.

It looks to me like this was a special case for the `Stats.Show` console command to default to `General` if you don't pass an argument. I changed the command implementation to handle that case properly, so this workaround shouldn't be needed anymore.

**Comment by Hoikas** [comment]
@dpogue, @colincornaby Have y'all had a chance to look at this? It has a lot of pipeline adjacent type stuff, so I'd like to get your input.

**Comment by dgelessus** [comment]
(rebased again to integrate #1578 and get off of the spicy version of liblzma)

**Review by dpogue — APPROVED** [review]

**Comment by colincornaby** [comment]
I'll take a look in the next few hours.

**Review by colincornaby — APPROVED** [review]
No questions right now. The Metal changes are (extremely) limited and I looked through the Direct3D changes and didn't see any concerns.

**Pull request merged**